autogen/agentchat/contrib/vectordb/pgvectordb.py

```diff
@@ -0,0 +1,710 @@
+                else:
+                    distance = self.euclidean_distance(fetched_document_array, vector)
+                results.append((fetched_document, distance))
+        cursor.close()
+        results = [results]
+        logger.debug(f"Query Results: {results}")
+        return results
+
+    @staticmethod
+    def convert_string_to_array(array_string) -> List[float]:
+        """
+        Convert a string representation of an array to a list of floats.
+
+        Parameters:
+        - array_string (str): The string representation of the array.
+
+        Returns:
+        - list: A list of floats parsed from the input string. If the input is
+          not a string, it returns the input itself.
+        """
+        if not isinstance(array_string, str):
+            return array_string
+        array_string = array_string.strip("[]")
+        array = [float(num) for num in array_string.split()]
+        return array
+
+    def modify(self, metadata, collection_name: str = None):
+        """
+        Modify metadata for the collection.
+
+        Args:
+            collection_name: The name of the collection.
+            metadata: The new metadata.
+
+        Returns:
+            None
+        """
+        if collection_name:
+            self.name = collection_name
+        cursor = self.client.cursor()
+        cursor.execute(
+            "UPDATE collections" "SET metadata = '%s'" "WHERE collection_name = '%s';", (metadata, self.name)
+        )
+        cursor.close()
+
+    def delete(self, ids: List[ItemID], collection_name: str = None):
+        """
+        Delete documents from the collection.
+
+        Args:
+            ids (List[ItemID]): A list of document IDs to delete.
+            collection_name (str): The name of the collection to delete.
+
+        Returns:
+            None
+        """
+        if collection_name:
+            self.name = collection_name
+        cursor = self.client.cursor()
+        cursor.execute(f"DELETE FROM {self.name} WHERE id IN ({ids});")
+        cursor.close()
+
+    def delete_collection(self, collection_name: str = None):
+        """
+        Delete the entire collection.
+
+        Args:
+            collection_name (Optional[str]): The name of the collection to delete.
+
+        Returns:
+            None
+        """
+        if collection_name:
+            self.name = collection_name
+        cursor = self.client.cursor()
+        cursor.execute(f"DROP TABLE IF EXISTS {self.name}")
+        cursor.close()
+
+    def create_collection(self, collection_name: str = None):
+        """
+        Create a new collection.
+
+        Args:
+            collection_name (Optional[str]): The name of the new collection.
+
+        Returns:
+            None
+        """
+        if collection_name:
+            self.name = collection_name
+        cursor = self.client.cursor()
+        cursor.execute(f"DROP TABLE IF EXISTS {self.name}")
```

Comment:
create collection will always drop an exisiting table?

Reply:
This will be removed.